service/src/main/java/bio/terra/workspace/app/controller/ControlledAzureResourceApiController.java

```diff
@@ -741,4 +746,102 @@ public ResponseEntity<ApiAzureManagedIdentityResource> getAzureManagedIdentity(
             .castByEnum(WsmResourceType.CONTROLLED_AZURE_MANAGED_IDENTITY);
     return new ResponseEntity<>(resource.toApiResource(), HttpStatus.OK);
   }
+
+  @Override
+  public ResponseEntity<ApiCreatedControlledAzureKubernetesNamespaceResult>
+      createAzureKubernetesNamespace(
+          UUID workspaceId, ApiCreateControlledAzureKubernetesNamespaceRequestBody body) {
+    features.azureEnabledCheck();
+
+    final AuthenticatedUserRequest userRequest = getAuthenticatedInfo();
+    final ControlledResourceFields commonFields =
+        toCommonFields(
+            workspaceId,
+            body.getCommon(),
+            landingZoneApiDispatch.getLandingZoneRegion(
+                userRequest, workspaceService.getWorkspace(workspaceId)),
+            userRequest,
+            WsmResourceType.CONTROLLED_AZURE_DATABASE);
+    workspaceService.validateMcWorkspaceAndAction(
+        userRequest, workspaceId, ControllerValidationUtils.samCreateAction(commonFields));
+    workspaceService.validateWorkspaceAndContextState(workspaceId, CloudPlatform.AZURE);
+
+    // append the workspace id to ensure the namespace is unique across all workspaces in the LZ
+    var kubernetesNamespace =
+        body.getAzureKubernetesNamespace().getNamespacePrefix() + "-" + workspaceId;
```

Comment:
Is there a possibility that this will end up being too long? (thinking of https://broadworkbench.atlassian.net/browse/WOR-1213)

Reply:
the prefix is limited to 25 characters, the workspace id is 36 so, including the dash, the namespace is limited to 62 which is not too long